broker/plugins/kubejobs/plugin.py

```diff
@@ -235,6 +266,39 @@ def errors(self):
             return ()
         return self.rds.lrange("job:errors", 0, -1)
 
+    def persist_state(self):
+        self.db_connector.\
+            put(self.app_id, self)
+
+    def synchronize(self):
+        """ Infer the job state from job status in Kubernetes.
+        If this job is active in Kubernetes, your state is ongoing.
+        If this job isn't active in Kubernetes, he can be completed
+        or failed.
+        If a exception has been throwed, this job not exists, so
+        your state is not found.
+
```

Comment:
"If an exception has been thrown, the job does not exist, so its state is 'not found'."